screeps-game-api/src/objects/impls/room.rs

```diff
@@ -99,6 +121,86 @@ impl Room {
     //     unimplemented!()
     // }
 
+    pub fn find_path<O, T, F>(
+        &self, 
+        from_pos: &O, 
+        to_pos: &T,
+        opts: FindOptions<F>,
+    ) -> Vec<Step> 
+    where
+        O: HasPosition,
+        T: HasPosition,
+        F: Fn(String, CostMatrix) -> (),
```

Comment:
The function should be able to return a `CostMatrix` in order to overwrite the one provided by the room- this probably wants to be `-> Option<CostMatrix>`.

Reply:
Honnestly, my first thought was not to support replacing the cost matrix. It is both inneficient and a pain to implement.